### PR TITLE
[Improvement] Refactor data parts

### LINKED ISSUE #1102: [improvement] replace the relationship between database and table and data parts from tbl_parts to parts field in Table struct
**Summary**

Replace the relationship between database and table and data parts from tbl_parts to parts field in Table struct

### PR DESCRIPTION
**Pull request by ariesdevil**
I hereby agree to the terms of the CLA available at: https://datafuse.rs/policies/cla/

## Summary

Minor refactoring to data_parts

## Changelog

- Improvement

## Related Issues

Fixes #1102 

## Test Plan

Unit Tests

Stateless Tests

### CODE REVIEW

File: fusestore/store/src/meta_service/state_machine.rs [hunk 1]
```diff
@@ -409,8 +409,9 @@ impl StateMachine {
     }
 
     pub fn get_data_parts(&self, db_name: &str, table_name: &str) -> Option<Vec<DataPartInfo>> {
-        let parts = self.tbl_parts.get(db_name);
-        parts.and_then(|m| m.get(table_name)).map(Clone::clone)
+        let db = self.databases.get(db_name).unwrap();
```
Comment (dantengsky) [inline fusestore/store/src/meta_service/state_machine.rs:412]: `unwrap` may not be a good idea, I guess you just forget to clean up it (them) :D

**Comment by drmingdrmer** [comment]
Stateless test didn't pass yet. 🤔 

It seems that unit tests does not covers these API if you have your local `cargo test` passed.

**Comment by ariesdevil** [comment]
> Stateless test didn't pass yet. 🤔
> 
> It seems that unit tests does not covers these API if you have your local `cargo test` passed.

Yup, I'll add these tests.

**Review by drmingdrmer — APPROVED** [review]
The codes look good and intuitive. Waiting for the CI to confirm the correctness

**Pull request merged**